app/javascript/components/data-store-form/data-store-form.schema.js

```diff
@@ -0,0 +1,64 @@
+import { componentTypes, validatorTypes } from '@@ddf';
+
+/** Namespace path field schema */
+const namespacePathField = () => ({
+  component: componentTypes.TEXT_FIELD,
+  id: 'namespacePath',
+  name: 'namespacePath',
+  label: __('Fully Qualified Name'),
+  isDisabled: true,
+});
+
+/** Name field schema */
+const nameField = (readOnly) => ({
+  component: componentTypes.TEXT_FIELD,
+  id: 'name',
+  name: 'name',
+  label: __('Name'),
+  maxLength: 128,
+  validate: [{ type: validatorTypes.REQUIRED }],
+  isRequired: true,
+  isDisabled: readOnly,
+});
+
+/** Description field schema */
+const descriptionField = (readOnly) => ({
+  component: componentTypes.TEXT_FIELD,
+  id: 'description',
+  name: 'description',
+  label: __('Description'),
+  maxLength: 128,
+  validate: [{ type: validatorTypes.REQUIRED }],
+  isDisabled: readOnly,
+});
+
+/** Enabled field schema */
+const enabledField = () => ({
+  component: componentTypes.CHECKBOX,
+  id: 'enabled',
+  name: 'enabled',
+  label: __('Enabled'),
+});
+
+/** Function to generate the form fields based on conditions */
+const generateFields = (domain, namespacePath, nameReadOnly, descReadOnly) => {
+  const fields = [];
+  if (!domain) { fields.push(namespacePathField(namespacePath)); }
+  fields.push(nameField(nameReadOnly));
+  fields.push(descriptionField());
+  if (domain) { fields.push(enabledField(descReadOnly)); }
```

Comment:
Is this right?  enabled_field doesn't seem to take a parameter.  Should this be descriptionField?  (I would have expected tests to catch this, so are there not tests for this case?)

Reply:
Or maybe descReadOnly is supposed to be passed to descriptionField on line 48?

Reply:
yes, the `descReadOnly` was supposed to be passed into the `descriptionField()`. it was last minute change and ended up in the wrong function.

This is not caught in test cases because the value for `readonly` is received as `undefined` in cases like this 
```
descriptionField()

/** Description field schema */
const descriptionField = (readOnly) => ({
  component: componentTypes.TEXT_FIELD,
  id: 'description',
  name: 'description',
  label: __('Description'),
  maxLength: 128,
  validate: [{ type: validatorTypes.REQUIRED }],
  isDisabled: readOnly,
});
```